code/modules/mob/living/basic/lavaland/watcher/watcher_overwatch.dm

```diff
@@ -116,13 +115,17 @@
 
 /datum/status_effect/overwatch/Destroy()
 	QDEL_NULL(link)
-	if (isnull(watcher))
-		return ..()
-	if (!overwatch_triggered) // Side effects in Destroy? Well it turns out `on_remove` is also just called on Destroy. But only if the owner isn't deleting.
+	if (!isnull(watcher))  // Side effects in Destroy? Well it turns out `on_remove` is also just called on Destroy. But only if the owner isn't deleting.
+		INVOKE_ASYNC(src, PROC_REF(unregister_watcher))
+
+	return ..()
+
+/// Clean up our association with the caster of this ability.
+/datum/status_effect/overwatch/proc/unregister_watcher()
+	if (!overwatch_triggered)
 		watcher.Stun(2 SECONDS, ignore_canstun = TRUE)
 	UnregisterSignal(watcher, list(COMSIG_QDELETING, COMSIG_LIVING_DEATH))
 	watcher = null
```

Comment:
oh i meant just move the `Stun()` to invoke async, everything else is _probably_ fine. if it's complicated just pass in `/mob/living/watcher` as an argument to `unregister_watcher()` like

```dm
if(!isnull(watcher))
     INVOKE_ASYNC(unregister_watcher(watcher)) // maybe this has to be a global instead? i don't think there are any side effects from my personal experience and i know the syntax is wrong this isn't meant to compile
    watcher = null
    UnregisterSignal(...)     
```

perhaps?